pkg/config/structure/unmarshal_test.go

```diff
@@ -40,6 +40,17 @@ type trapsConfig struct {
 	Namespace        string   `yaml:"namespace"`
 }
 
+// We don't use config mock here to not create cycle dependencies (same reason why config mock are not used in
+// pkg/config/{setup/model})
```

Comment:
Cool, this explains well why this fix works to handle the issue I originally ran into when I last tried to use UnmarshalKey in the remaining sites.